components/core/src/package/archive.rs

```diff
@@ -417,6 +419,86 @@ pub trait FromArchive: Sized {
     fn from_archive(archive: &mut PackageArchive) -> result::Result<Self, Self::Error>;
 }
 
+// Exposes the extended archive and header metadata. Types are stored as non habitat primitives
+// with the intent being ease of deserialization into content such as conveniently formatted json
+// at the client display layer.
+#[derive(Deserialize, Serialize)]
+pub struct PackageArchiveInfo {
```

Comment:
What do we gain by keeping these `String`s instead of habitat types? I worry that relying on the `ToString` trait instead of `Serialize` could lead to inconsistencies in how data is displayed. If we want to keep everything as `String`s could we use `serde::to_string` intead of the `ToString` trait?

Reply:
What's the relationship between `ToString` and `serde::to_string`? I thought the latter took advantage of the former?

One problem we've historically had is that a lot of our types have uncritically derived `Serialize` and `Deserialize`, which often yields less-than-ideal results. In particular, package identifiers get converted into an Object with keys for each component of the identifier (i.e., `origin`, `package`, etc.). Every time I've had to manipulate JSON derived from such structures, I've always had to piece together an identifier from the components, rather than just getting the identifier directly (see https://github.com/habitat-sh/habitat/pull/7814#discussion_r456566956). So that's one instance where sticking to the "naive" JSON implementations we have doesn't really serve the use case we're trying to support.

To the extent possible, maintaining consistent usage of core Habitat types everywhere we can would be nice, with targeted behavioral overrides when necessary (which, I think, is what you're getting at here, right @davidMcneil?) If we keep seeing the same "targeted behavioral overrides" enough, we could start refactoring that back into the original type. If we can do that, but still get an end result that "feels right" and works for the user, then :partying_face: :tada:

Reply:
> What's the relationship between `ToString` and `serde::to_string`? I thought the latter took advantage of the former?

Yes, it can. Although, they can be distinct as well. `PackageIdent` is an example of this. It probably does not apply in this case, but I was worried about having some output use `Serialize` and some use `ToString` and the potential inconsistency between those. In this case, it sounds like we explicitly want `ToString` at least for the `PackageIdent`.

Reply:
> One problem we've historically had is that a lot of our types have uncritically derived Serialize and Deserialize, which often yields less-than-ideal results. In particular, package identifiers get converted into an Object with keys for each component of the identifier (i.e., origin, package, etc.)

This is the exact problem I'm trying to solve, hence the naive `String` types to hold custom formatting which then `serde_json` serializes into more pleasant json. Maybe there's a better way to accomplish this @davidMcneil by still preserving the underlying Habitat Types, yet converting them into a format that the end user expects to consume?

Reply:
@jeremymv2 That makes sense. There is probably not a better way to accomplish it outside of changing the underlying `Serialization` implementation. This would probably be worth exploring but is outside the scope of this PR.